From f17e62921e5f8ad3a66bd5e74053046287ffcba4 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Thu, 4 Apr 2019 16:32:21 +0800
Subject: [PATCH 11/30] drm/bridge: nwl-dsi: fix compile error

In upstream commit 9928688492ae ("drm/bridge: change return type
of drm_bridge_add function"), the function drm_bridge_add has no
return value anymore, so remove the return value of it to avoid
compile error.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 drivers/gpu/drm/bridge/nwl-dsi.c  | 6 ++----
 drivers/gpu/drm/imx/nwl_dsi-imx.c | 7 +------
 2 files changed, 3 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/bridge/nwl-dsi.c b/drivers/gpu/drm/bridge/nwl-dsi.c
index 8cad5f4dabbb..4a0ab74a675d 100644
--- a/drivers/gpu/drm/bridge/nwl-dsi.c
+++ b/drivers/gpu/drm/bridge/nwl-dsi.c
@@ -1108,7 +1108,7 @@ static int nwl_dsi_probe(struct platform_device *pdev)
 	struct nwl_mipi_dsi *dsi;
 	struct clk *clk;
 	struct resource *res;
-	int ret;
+	int ret = 0;
 
 	dsi = devm_kzalloc(dev, sizeof(*dsi), GFP_KERNEL);
 	if (!dsi)
@@ -1181,9 +1181,7 @@ static int nwl_dsi_probe(struct platform_device *pdev)
 	dsi->bridge.funcs = &nwl_dsi_bridge_funcs;
 	dsi->bridge.of_node = dev->of_node;
 
-	ret = drm_bridge_add(&dsi->bridge);
-	if (ret < 0)
-		dev_err(dev, "Failed to add nwl-dsi bridge (%d)\n", ret);
+	drm_bridge_add(&dsi->bridge);
 
 	return ret;
 }
diff --git a/drivers/gpu/drm/imx/nwl_dsi-imx.c b/drivers/gpu/drm/imx/nwl_dsi-imx.c
index 37317ceee0af..85bff15bd4f6 100644
--- a/drivers/gpu/drm/imx/nwl_dsi-imx.c
+++ b/drivers/gpu/drm/imx/nwl_dsi-imx.c
@@ -973,12 +973,7 @@ static int imx_nwl_dsi_probe(struct platform_device *pdev)
 		dsi->bridge.funcs = &imx_nwl_dsi_bridge_funcs;
 		dsi->bridge.of_node = np;
 
-		ret = drm_bridge_add(&dsi->bridge);
-		if (ret) {
-			dev_err(dev, "Failed to add imx-nwl-dsi bridge (%d)\n",
-				ret);
-			return ret;
-		}
+		drm_bridge_add(&dsi->bridge);
 		dev_info(dev, "Added drm bridge!");
 		return 0;
 	}
-- 
2.17.1

